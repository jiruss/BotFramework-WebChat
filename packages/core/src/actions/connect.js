const CONNECT = 'DIRECT_LINE/CONNECT';
const CONNECT_PENDING = `${ CONNECT }_PENDING`;
const CONNECT_REJECTED = `${ CONNECT }_REJECTED`;
const CONNECT_FULFILLING = `${ CONNECT }_FULFILLING`;
const CONNECT_FULFILLED = `${ CONNECT }_FULFILLED`;

export default function ({ directLine, userID }) {
  return {
    type: CONNECT,
    payload: { directLine, userID }
  };
}

export {
  CONNECT,
  CONNECT_PENDING,
  CONNECT_REJECTED,
  CONNECT_FULFILLING,
  CONNECT_FULFILLED
}
