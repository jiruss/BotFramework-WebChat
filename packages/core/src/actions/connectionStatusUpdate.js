const CONNECTION_STATUS_UPDATE = 'DIRECT_LINE/CONNECTION_STATUS_UPDATE';

export default function (connectionStatus) {
  return {
    type: CONNECTION_STATUS_UPDATE,
    payload: { connectionStatus }
  };
}

export { CONNECTION_STATUS_UPDATE }
