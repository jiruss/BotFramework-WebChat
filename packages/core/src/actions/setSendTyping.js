const SET_SEND_TYPING = 'WEB_CHAT/SET_SEND_TYPING';

export default function (value) {
  return {
    type: SET_SEND_TYPING,
    payload: { sendTyping: !!value }
  };
}

export { SET_SEND_TYPING }
