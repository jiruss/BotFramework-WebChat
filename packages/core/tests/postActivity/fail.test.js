import createFacility from '../createFacility';

import { POST_ACTIVITY, POST_ACTIVITY_REJECTED } from '../../src/actions/postActivity';
import * as ActivityClientState from '../../src/constants/ActivityClientState';

test('Post activity failed', async () => {
  const { actions, directLine, latestActions, store } = await createFacility();

  // Clear out actions
  actions.splice(0);

  store.dispatch({
    type: POST_ACTIVITY,
    payload: {
      activity: {
        type: 'message',
        text: 'Hello, World!'
      }
    }
  });

  await 0;
  expect(directLine.pendingPosts.map(post => post.peek())).toMatchSnapshot('pending to send: activity');
  expect(store.getState().activities).toMatchSnapshot('pending to send: store');
  expect(store.getState().activities[0]).toHaveProperty('channelData.state', ActivityClientState.SENDING);
  expect(store.getState().activities[0]).not.toHaveProperty('id');

  directLine.pendingPosts[0].error('some error');

  await 0;
  expect(actions).toMatchSnapshot('send failed: actions');
  expect(store.getState().activities).toMatchSnapshot('send failed: store');

  expect(latestActions[POST_ACTIVITY_REJECTED]).toHaveProperty('error', true);
  expect(latestActions[POST_ACTIVITY_REJECTED]).toHaveProperty('payload', 'some error');
  expect(store.getState().activities[0]).toHaveProperty('channelData.state', ActivityClientState.SEND_FAILED);
  expect(store.getState().activities[0]).not.toHaveProperty('id');
});
