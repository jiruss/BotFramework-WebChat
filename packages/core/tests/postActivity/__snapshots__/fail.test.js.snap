// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post activity failed: pending to send: activity 1`] = `
Array [
  Object {
    "channelData": Object {
      "clientActivityID": "4feornb",
    },
    "from": Object {
      "id": "default-user",
      "name": "John Doe",
    },
    "text": "Hello, World!",
    "timestamp": "2000-01-01T00:00:01.000Z",
    "type": "message",
  },
]
`;

exports[`Post activity failed: pending to send: store 1`] = `
Array [
  Object {
    "channelData": Object {
      "clientActivityID": "4feornb",
      "state": "sending",
    },
    "from": Object {
      "id": "default-user",
      "name": "John Doe",
    },
    "text": "Hello, World!",
    "timestamp": "2000-01-01T00:00:01.000Z",
    "type": "message",
  },
]
`;

exports[`Post activity failed: send failed: actions 1`] = `
Array [
  Object {
    "payload": Object {
      "activity": Object {
        "text": "Hello, World!",
        "type": "message",
      },
    },
    "type": "DIRECT_LINE/POST_ACTIVITY",
  },
  Object {
    "meta": Object {
      "clientActivityID": "4feornb",
    },
    "type": "DIRECT_LINE/POST_ACTIVITY_PENDING",
  },
  Object {
    "payload": Object {
      "activity": Object {
        "channelData": Object {
          "clientActivityID": "4feornb",
          "state": "sending",
        },
        "from": Object {
          "id": "default-user",
          "name": "John Doe",
        },
        "text": "Hello, World!",
        "timestamp": "2000-01-01T00:00:01.000Z",
        "type": "message",
      },
    },
    "type": "DIRECT_LINE/INCOMING_ACTIVITY",
  },
  Object {
    "error": true,
    "meta": Object {
      "clientActivityID": "4feornb",
    },
    "payload": "some error",
    "type": "DIRECT_LINE/POST_ACTIVITY_REJECTED",
  },
]
`;

exports[`Post activity failed: send failed: store 1`] = `
Array [
  Object {
    "channelData": Object {
      "clientActivityID": "4feornb",
      "state": "send failed",
    },
    "from": Object {
      "id": "default-user",
      "name": "John Doe",
    },
    "text": "Hello, World!",
    "timestamp": "2000-01-01T00:00:01.000Z",
    "type": "message",
  },
]
`;
